stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/channel/add/group/select_name/AddGroupChannelSelectNameMembersAdapter.kt

```diff
@@ -0,0 +1,53 @@
+package io.getstream.chat.ui.sample.feature.channel.add.group.select_name
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.client.models.name
+import io.getstream.chat.ui.sample.databinding.AddGroupChannelSelectNameMemberItemBinding
+import io.getstream.chat.ui.sample.feature.channel.add.group.select_name.AddGroupChannelSelectNameMembersAdapter.DeleteMemberClickListener
+
+class AddGroupChannelSelectNameMembersAdapter :
+    ListAdapter<User, AddGroupChannelSelectNameMembersAdapter.MemberViewHolder>(
+        object : DiffUtil.ItemCallback<User>() {
+            override fun areItemsTheSame(oldItem: User, newItem: User): Boolean {
+                return oldItem.id == newItem.id
+            }
+
+            override fun areContentsTheSame(oldItem: User, newItem: User): Boolean {
+                return oldItem == newItem
+            }
+        }
+    ) {
+
+    var deleteMemberClickListener: DeleteMemberClickListener = DeleteMemberClickListener { }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MemberViewHolder {
+        return AddGroupChannelSelectNameMemberItemBinding
+            .inflate(LayoutInflater.from(parent.context), parent, false)
+            .let { MemberViewHolder(it, deleteMemberClickListener) }
+    }
+
+    override fun onBindViewHolder(holder: MemberViewHolder, position: Int) {
+        holder.bind(getItem(position))
+    }
+
+    class MemberViewHolder(
+        private val binding: AddGroupChannelSelectNameMemberItemBinding,
+        private val deleteMemberClickListener: DeleteMemberClickListener
```

Comment:
The listener passed in as a parameter here will get outdated if the Adapter's listener is changed. We could make this an `inner class` instead that uses the Adapter's listener directly to avoid that.